src/blog/2024/01/send-a-file.md

```diff
@@ -0,0 +1,162 @@
+---
+title: Send a File to Node-RED
+subtitle: A guide to sending a csv file to Node-RED and start interacting with it.
+date: 2024-01-04
+authors: ["grey-dziuba"]
+image: /blog/2024/01/images/send-file-to-node-red.png
+tags:
+    - posts
+    - flowfuse
+    - how-to
+---
+
+Have you ever needed to send a CSV file to your Node-RED instance? This file can go on to populate a shift schedule, product specifications, or some other configuration file that is used. In this guide, we are leveraging a Python script that runs a Flask application to post the file to a Node-RED instance and organize the data to be sent on or used.
+<!--more-->
+
+# Why would you need to send a file to Node-RED?
+
+Often times it is necessary to update lookup tables in a SQL database, but you don't necessarily want to give access to everyone to edit the database, nor do you want to have to do it all yourself. This can often be seen when new products are introduced into a manufacturing facility. It may not be often, but enough that it warrants its own application. This process will guide you in a way that will enable your teammates to upload the files to the system themselves.
+
+# Summary
+
+There are many approaches that can be taken when solving this. We are going to go over 4. The first is the most simple approach but very manual.  A simple cURL command to send the file.  The second is to leverage FlowFuse's Dashboards to allow the user to navigate to the webpage upload a file and visualize before sending it on to the next process.  The third approach is a simple script that will be shared below. It is a simple Python application that allows the user to send a file with a simple command, but this might require a little more technical skills that the end user may not feel comfortable with. Which brought us to last option. A web-based application that allows the user to upload files to a browser with a selectable endpoint. All have their pros and cons. I will let you decide which is best for your application.
```

Comment:
```suggestion
There are many approaches that can be taken when solving this. We are going to go over 4. The first is the most simple approach but very manual. A simple cURL command to send the file. The second is to leverage [FlowFuse's Dashboard](https://dashboard.flowfuse.com/) to allow the user to navigate to the webpage upload a file and visualize before sending it on to the next process. The third approach is a simple script that will be shared below. It is a simple Python application that allows the user to send a file with a simple command, but this might require a little more technical skills that the end user may not feel comfortable with. Which brought us to last option. A web-based application that allows the user to upload files to a browser with a selectable endpoint. All have their pros and cons. I will let you decide which is best for your application.
```